File: php/profiling/profiling-db.php
<?php
error_reporting(E_ALL);

require "credentials.php";
require "page.php";

function light_query() {
  global $db;
  $query = "
    SELECT * 
    FROM table1, table2
  ";
  $result = pg_query($db, $query);
  while ($row = pg_fetch_assoc($result)) {
    
  }
}

function heavy_query() {
  global $db;
  
  // table1 does not contain an index for field1
  $query = "
    SELECT * 
    FROM table1, table2
    ORDER BY table1.field1";

  $result = pg_query($db, $query);
  while ($row = pg_fetch_assoc($result)) {
    
  }
}

function mid_tier() {
  light_query();
  heavy_query();
}

function top_tier() {
  mid_tier();
}

page_head();

$db = pg_connect(
    "host=" . DB_HOST  .
    " dbname=" . DB_NAME  . 
    " user=" . DB_USER .
    " password=" . DB_PASS)
    or die('Could not connect: ' . pg_last_error());

top_tier();

page_footer("db");
?>

File: python/profiling/profiling-db.py
import psycopg2

import credentials

def light_query():
    st = """
        SELECT *
        FROM table1, table2
    """
    cur.execute(st)
    for row in cur:
        pass

def heavy_query():
    st = """
        SELECT *
        FROM table1, table2
        ORDER BY table1.field1
    """
    cur.execute(st)
    for row in cur:
        pass

def mid_tier():
    light_query()
    heavy_query()

def top_tier():
    mid_tier()

conn = psycopg2.connect(
        database=credentials.DB_NAME, 
        user=credentials.DB_USER,
        host=credentials.DB_HOST,
        password=credentials.DB_PASS
)

cur = conn.cursor()

top_tier()

